Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Utilities/RotateToValue.cs

```diff
@@ -1,97 +1,198 @@
 using UnityEngine;
 using UnityEngine.Events;
 
-namespace HoloToolkit.Examples.Prototyping
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Utilities
 {
     /// <summary>
-    /// animates the rotation of an object with eases
+    /// Animates the rotation of an object with easing's
     /// </summary>
     public class RotateToValue : MonoBehaviour
     {
-        /// <summary>
-        /// ease types
-        ///     Linear: steady progress
-        ///     EaseIn: ramp up in speed
-        ///     EaseOut: ramp down in speed
-        ///     EaseInOut: ramp up then down in speed
-        ///     Free: super ease - just updates as the TargetValue changes
-        /// </summary>
-        public enum LerpTypes { Linear, EaseIn, EaseOut, EaseInOut, Free }
+        #region Properties and fields
 
         [Tooltip("The object to animate")]
-        public GameObject TargetObject;
+        [SerializeField]
+        private GameObject targetObject;
+
+        public GameObject TargetObject
+        {
+            get { return targetObject; }
+            set { targetObject = value; }
+        }
 
         [Tooltip("The rotation value to animate to")]
-        public Quaternion TargetValue;
+        [SerializeField]
+        private Quaternion targetValue;
+
+        public Quaternion TargetValue
+        {
+            get { return targetValue; }
+            set { targetValue = value; }
+        }
 
         [Tooltip("The type of ease to apply to the tween")]
-        public LerpTypes LerpType;
+        [SerializeField]
+        private EasingTypes lerpType;
+
+        public EasingTypes LerpType
+        {
+            get { return lerpType; }
+            set { lerpType = value; }
+        }
 
         [Tooltip("Duration of the animation in seconds")]
-        public float LerpTime = 1f;
+        [SerializeField]
+        private float lerpTime = 1f;
+
+        public float LerpTime
+        {
+            get { return lerpTime; }
+            set { lerpTime = value; }
+        }
 
-        [Tooltip("auto start? or status")]
-        public bool IsRunning = false;
+        [Tooltip("Is the Easing function running?")]
+        [SerializeField]
+        private bool isRunning = false;
+
+        public bool IsRunning
+        {
+            get { return isRunning; }
+            set { isRunning = value; }
+        }
 
         [Tooltip("Use the localRotation instead of world rotation")]
-        public bool ToLocalTransform = false;
+        [SerializeField]
+        private bool toLocalTransform = false;
+
+        public bool ToLocalTransform
+        {
+            get { return toLocalTransform = false; }
+            set { toLocalTransform = value; }
+        }
 
-        [Tooltip("animation complete!")]
-        public UnityEvent OnComplete;
+        [Tooltip("Animation complete!")]
+        [SerializeField]
+        private UnityEvent onComplete;
+
+        public UnityEvent OnComplete
```

Comment:
Is there a way to also raise an event through the MRTK's event system, or event as an Action?

Reply:
Let's review that in Beta.  I expect the examples scenes to grow moving forward.